features/FEATURE_BLE/targets/TARGET_NORDIC/TARGET_NRF5/source/nRF5xGattServer.cpp

```diff
@@ -451,6 +514,140 @@ void nRF5xGattServer::hwCallback(ble_evt_t *p_ble_evt)
+                    }
+
+                    // initialize the first request by setting the offset
+                    if (req->length == 0) {
+                        req->attr_handle = input_req.handle;
+                        req->offset = input_req.offset;
+                    }
+
+                    // it is disalowed to write backward
+                    if (input_req.offset < req->offset) {
+                        sd_ble_gatts_rw_authorize_reply(conn_handle, &write_auth_invalid_offset_reply);
+                        releaseLongWriteRequest(conn_handle);
+                        return;
+                    }
+
+                    // it should be the subsequent write
+                    if ((req->offset + req->length) != input_req.offset) {
+                        sd_ble_gatts_rw_authorize_reply(conn_handle, &write_auth_invalid_offset_reply);
+                        releaseLongWriteRequest(conn_handle);
+                        return;
+                    }
+
+                    // it is not allowed to write multiple characteristic with the same request
+                    if (input_req.handle != req->attr_handle) {
+                        sd_ble_gatts_rw_authorize_reply(conn_handle, &write_auth_invalid_reply);
+                        releaseLongWriteRequest(conn_handle);
+                        return;
+                    }
+
+                    // start the copy of what is in input
+                    memcpy(req->data + req->length, input_req.data, input_req.len);
+
+                    // update the lenght of the data written
+                    req->length = req->length + input_req.len;
+
+                    // success, signal it to the softdevice
+                    ble_gatts_rw_authorize_reply_params_t reply = {
+                        .type = BLE_GATTS_AUTHORIZE_TYPE_WRITE,
+                        .params = {
+                            .write = {
+                                .gatt_status = BLE_GATT_STATUS_SUCCESS,
+                                .update = 1,
+                                .offset = input_req.offset,
+                                .len = input_req.len,
+                                .p_data = input_req.data
+                            }
+                        }
+                    };
+
+                    sd_ble_gatts_rw_authorize_reply(conn_handle, &reply);
+                }   return;
+
+                case BLE_GATTS_OP_EXEC_WRITE_REQ_CANCEL: {
+                    releaseLongWriteRequest(conn_handle);
+                    sd_ble_gatts_rw_authorize_reply(conn_handle, &write_auth_succes_reply);
+                }   return;
+
+                case BLE_GATTS_OP_EXEC_WRITE_REQ_NOW: {
+                    long_write_request_t* req = findLongWriteRequest(conn_handle);
+                    if (!req) {
+                        sd_ble_gatts_rw_authorize_reply(conn_handle, &write_auth_invalid_reply);
+                        releaseLongWriteRequest(conn_handle);
```

Comment:
This call is unnecessary. From the line 595 it is known that there is no such write request.